src/pip/_internal/network/download.py

```diff
@@ -81,7 +81,7 @@ def parse_content_disposition(content_disposition: str, default_filename: str) -
     Parse the "filename" value from a Content-Disposition header, and
     return the default filename if the result is empty.
     """
-    _type, params = cgi.parse_header(content_disposition)
+    _type, params = parse_dict_header(content_disposition)
```

Comment:
Hmm, `parse_dict_header` is not listed in Requests’s documentation (as far as I can tell), so we might need to make sure it’s OK to use it.

Reply:
https://github.com/psf/requests/issues/4931#issuecomment-452000709

Reply:
Thanks both of you, looks like parse_dict_header() may not be intended as a public API after all (also I misgauged the return value anyway, so it would have needed more work).